File: projects/javascript/10-http-server/starter/main.js
// 10-http-server (starter)
const readline = require('readline');

// TODO(step-1): HTTP 요청 문자열 파서
function parseRequest(raw) {
  return { method: 'GET', path: '/' };
}

// TODO(step-2): 라우터 클래스
class Router {
  #routes = [];
  add(method, path, handler) {}
  match(method, path) { return null; }
}

// TODO(step-3): 응답 포맷팅 유틸리티
function buildResponse(status, body, contentType = 'text/plain') {
  return "";
}

// TODO(step-4): 미들웨어 실행기
function runMiddlewares(req, middlewares, handler) {
  return handler(req);
}

// TODO(step-5): HTTP 스트림 처리
const rl = readline.createInterface({ input: process.stdin });
rl.on('line', () => {});

File: projects/rust/10-http-server/starter/main.rs
// 10-http-server (starter)
#![allow(unused_variables, dead_code, unused_mut, unused_imports)]
use std::io::{self, BufRead};

// TODO(step-1): HttpRequest 구조체
struct HttpRequest {
    method: String,
    path: String,
}

// TODO(step-2): parse_request 요청 파서
fn parse_request(raw: &str) -> Option<HttpRequest> {
    None
}

// TODO(step-3): route 핸들러
fn route(req: &HttpRequest) -> String {
    String::new()
}

// TODO(step-4): HttpResponse 포맷팅
fn format_response(status: u16, body: &str) -> String {
    String::new()
}

// TODO(step-5): REPL CLI
fn main() {
    let stdin = io::stdin();
    for line in stdin.lock().lines() {}
}
